tools/toolchain/gen_mksurf_namelist.py

```diff
@@ -0,0 +1,658 @@
+#!/usr/bin/env python
+
+# 2020-11-08                Negin Sobhani
+
+"""
+|------------------------------------------------------------------|
+|---------------------  Instructions  -----------------------------|
+|------------------------------------------------------------------|
+This Python script is part of the simplified toolchain for creating
+the surface dataset for ctsm cases.
+This script should be used as the first step of the new toolchain. 
+It will automatically creating namelist (control  file) that is 
+needed for creating surface dataset and other relevant files for
+running CTSM cases. 
+For transient cases, it will also create a txt file that includes the
+landuse files for every year. 
+
+-------------------------------------------------------------------
+Instructions for running on Cheyenne/Casper:
+
+load the following into your local environment:
+
+    module load python
+    ncar_pylib
+-------------------------------------------------------------------
+To see the available options:
+    ./gen_mksurf_namelist.py --help
+
+To run the script:
+    ./gen_mksurf_namelist.py
+ 
+To remove NPL from your environment on Cheyenne/Casper:
+    deactivate
+-------------------------------------------------------------------
+"""
+
+#TODO (NS)
+
+# -[x] Add default values in the help page.
+# -[x] Add info for help page note for end_year -- by default is start_year
+# -[ ] Possibly remove year --years and range options
+#      Currently comment them out.
+
+# -[ ] maybe a verbose option and removing debug
+# -[x] --debug mode is not working...
+
+# -[ ] add error check for hi-res and years if they are 1850 and 2005.
+
+# -[x] different path for each range of years for transient cases. 
+#      default should be picked based on the year. 1850 - 2015 -->
+#       /glade/p/cesm/cseg/inputdata/lnd/clm2/rawdata/pftcftdynharv.0.25x0.25.LUH2.histsimyr1850-2015.c170629/
+#      850-1850 --> 
+#       pftcftdynharv.0.25x0.25.LUH2.histsimyr0850-1849.c171012
+
+# -[ ] hirespft data only for 2005?
+
+#QUESTIONS:
+# -[ ] Do we need --crop to accept y/n or just --crop is enough?
```

Comment:
At the moment I'm leaning in favor of just --crop
because that's consistent with other options, such as --vic --hirespft --fast --glc
It may confuse users if only crop requires a y/n.

Reply:
We discussed this for subset_data in a CSEG meeting. The consensus is to use --feature and --nofeature instead of --feature [y/n]. Here the crop flag is on by default so I introduced ---nocrop option.

Reply:
The question for @ekluzek and others is that do we want crop to be on by default or off? 

I can easily change this to default no-crop which translate to having --crop option instead.

Reply:
The default should be crop on, since you can turn it off in CTSM. So almost all of the datasets we create will be with crop on. There will only be some exceptions to that where we'll create a dataset with crop off (one I can think of offhand are the VIC datasets). And a user might want to turn crop off explicitly.